purldb-toolkit/README.rst

```diff
@@ -20,84 +25,541 @@ Use this command to get basic help::
       --help             Show this message and exit.
 
+Examples
+########
+
+**Submit multiple PURLs using the command line:**
+
+.. code-block:: none
+
+    purlcli validate --purl pkg:npm/canonical-path@1.0.0 --purl pkg:nginx/nginx@0.8.9 --output <path/to/output.json>
+
+*Sample output:*
+
+.. code-block:: console
+
+    {
+        "headers": [
+            {
+                "tool_name": "purlcli",
+                "tool_version": "0.2.0",
+                "options": {
+                    "command": "validate",
+                    "--purl": [
+                        "pkg:npm/canonical-path@1.0.0",
+                        "pkg:nginx/nginx@0.8.9"
+                    ],
+                    "--file": null,
+                    "--output": "<path/to/output.json>"
+                },
+                "errors": [],
+                "warnings": [
+                    "'check_existence' is not supported for 'pkg:nginx/nginx@0.8.9'"
+                ]
+            }
+        ],
+        "packages": [
+            {
+                "purl": "pkg:npm/canonical-path@1.0.0",
+                "valid": true,
+                "exists": true,
+                "message": "The provided Package URL is valid, and the package exists in the upstream repo."
+            },
+            {
+                "purl": "pkg:nginx/nginx@0.8.9",
+                "valid": true,
+                "exists": null,
+                "message": "The provided PackageURL is valid, but `check_existence` is not supported for this package type."
+            }
+        ]
+    }
+
+
+**Submit multiple PURLs using a .txt file:**
+
+.. code-block:: none
+
+    purlcli validate --file <path/to/output.txt> --output <path/to/output.json>
+
+*Sample input.txt:*
+
+.. code-block:: console
+
+    pkg:npm/canonical-path@1.0.0
+    pkg:nginx/nginx@0.8.9
+
+
+Notes
+#######
+
+``validate`` calls the ``public.purldb.io/api/validate/`` endpoint.
+
+
+----
+
+
+'versions' -- collect package versions for a PURL
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: none
 
-- Collect package versions for a PURL::
-      
     $ purlcli versions  --help
     Usage: purlcli versions [OPTIONS]
-    
+
       Given one or more PURLs, return a list of all known versions for each PURL.
-    
-      Version information is not needed in submitted PURLs and if included will be
-      removed before processing.
-    
+
     Options:
       --purl TEXT        PackageURL or PURL.
       --output FILENAME  Write versions output as JSON to FILE.  [required]
       --file FILENAME    Read a list of PURLs from a FILE, one per line.
       --help             Show this message and exit.
 
+Examples
+########
+
+**Submit multiple PURLs using the command line:**
+
+.. code-block:: none
+
+    purlcli versions --purl pkg:npm/canonical-path --purl pkg:nginx/nginx --output <path/to/output.json>
+
+*Sample output:*
+
+.. code-block:: console
```

Comment:
```suggestion
.. code-block:: json
```

Reply:
Consider the same for all applicable JSON snippets

Reply:
All blocks updated as needed to `.. code-block:: console`, `.. code-block:: json` or `.. code-block:: text`.